src/metabase/models/params/chain_filter.clj

```diff
@@ -634,12 +627,24 @@
       []))
 
 (defn- search-cached-field-values? [field-id constraints]
-  (and (use-cached-field-values? field-id constraints true)
-       (isa? (db/select-one-field :base_type Field :id field-id) :type/Text)))
+  (and (use-cached-field-values? field-id)
+       (isa? (db/select-one-field :base_type Field :id field-id) :type/Text)
+       (db/exists? FieldValues (merge {:field_id field-id, :values [:not= nil], :human_readable_values nil}
```

Comment:
for search we need to be a bit more strict here because if we want to search from `FieldValues`, we need to make sure the FieldValues contains all possible values (has_more_values=false) otherwise we should run an MBQL to search.